SharpSource/SharpSource/Diagnostics/ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzer.cs

```diff
@@ -118,7 +125,19 @@ or TypeKind.Array
 
         if (!implementsEquals || !implementsGetHashCode)
         {
-            context.ReportDiagnostic(Diagnostic.Create(Rule, argument.Syntax.GetLocation(), argumentType.Name));
+            var syntaxNode = argument?.Syntax ?? extensionMethodInvocationWithoutArguments!.Syntax;
```

Comment:
Can you rewrite this bit so we don't use the suppression operator?

Reply:
I tried,but compiler ignores that either argument or extensionMethodInvocationWithoutArguments is not null.
[Example on SharpLab](https://sharplab.io/#v2:C4LgbgpgTglgZgTwDQBMQGoA+ABAzAAmwCZ8BhfAbwFgAofewg7ARgAZ8BZAChdYH58AQySE2AgEYBKSrQZz88fF0H4AvKvwA7AK4AbXfgBkh/OLUad+ydTry72AOyjWAOgCiAWwAOwBAG56WTt8AF8gu3D5RyF8PgkA4PlIsJoQoA==)

Reply:
I think something like this will work:

```cs
var syntaxNode = argument?.Syntax ?? extensionMethodInvocationWithoutArguments?.Syntax;
if (syntaxNode is not null)
{
    context.ReportDiagnostic(Diagnostic.Create(Rule, syntaxNode.GetLocation(), argumentType.Name));
}
```

I'm happy to take the additional check if it means we don't need to suppress nullability analysis

Reply:
Ok